### ISSUE TITLE
Inconsistent result between NumPy and PyTorch in complex power for 0 

### ISSUE THREAD

**Issue by Kiyosora** [pytorch/pytorch#51714]
## 🐛 Bug
When calculating 0 to power a complex number with negative image part, got an inconsistent result with NumPy.
I'm not sure whether this inconsistency is in our favor, I could have a try if a fixing is needed.

## To Reproduce
```
>>> (0)**np.array([1-1j])
array([nan+nanj])
>>> (0)**torch.tensor([1-1j])
tensor([0.+0.j])
```

## Expected behavior
Consistent complex results with NumPy, like:
```
>>> (0)**torch.tensor([1-1j])
tensor([nan+nanj])
```

## Environment
```
Collecting environment information...
PyTorch version: 1.8.0a0+d3b3387
Is debug build: True
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A
OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 7.5.0
Clang version: Could not collect
CMake version: version 3.14.0
Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1050 Ti
Nvidia driver version: 450.51.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.8.0a0
[conda] blas                      1.0                         mkl    defaults
[conda] magma-cuda102             2.5.2                         1    pytorch
[conda] mkl                       2020.1                      217    defaults
[conda] mkl-include               2020.1                      217    defaults
[conda] mkl-service               2.3.0            py37he904b0f_0    defaults
[conda] mkl_fft                   1.1.0            py37h23d657b_0    defaults
[conda] mkl_random                1.1.1            py37h0573a6f_0    defaults
[conda] numpy                     1.18.5           py37ha1c710e_0    defaults
[conda] numpy-base                1.18.5           py37hde5b4d6_0    defaults
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] torch                     1.8.0a0                   dev_0    <develop>
```

**Comment by ezyang** [comment]
It seems like we have a number of these NaN discrepancies on complex, see also https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3A%22module%3A+NaNs+and+Infs%22+label%3A%22module%3A+complex%22

**Comment by ngimel** [comment]
0**z where Re(z)>0 is 0, so we are correct here.

**Comment by anjali411** [comment]
> 0**z where Re(z)>0 is 0, so we are correct here.

yeah! this https://math.stackexchange.com/questions/2087100/is-zero-to-the-power-of-an-imaginary-number-still-zero stack exchange answer gives a detailed reasoning using limits to show that the correct answer in this cases is indeed 0. 
Filed an issue on numpy to bring this bug to notice: https://github.com/numpy/numpy/issues/18378.

As for the other cases (when Re(z) <= 0 where z is the exponent), we are consistent with numpy when Re(z) = 0:
```py 
>>> (0)**np.array([1j])
array([nan+nanj])
>>> (0)**torch.tensor([1j])
tensor([nan+nanj])
>>> (0)**torch.tensor([-1j])
tensor([nan+nanj])
>>> (0)**np.array([-1j])
array([nan+nanj])
```
However, inconsistent for the case when `Re(z) < 0`:
```py
>>> (0)**torch.tensor([-1-1j])
tensor([inf+nanj])
>>> (0)**np.array([-1-1j])
array([nan+nanj])
```

**Comment by cpuhrsch** [comment]
Seems related to https://github.com/pytorch/pytorch/issues/48000
